File: Marmalade/MarmaladeODK/source/android/CallbacksSetDeveloperId.cpp
/*
 * Copyright (C) 2012, 2013 OUYA, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

#include "CallbacksSetDeveloperId.h"
#include "CallbackSingleton.h"

#include "ODK_internal.h"
#include "s3eEdk.h"
#include "s3eEdk_android.h"
#include "IwDebug.h"

#include <stdio.h>

void CallbacksSetDeveloperId::RegisterCallback(s3eCallback callback, s3eCallback* savedCallback, int callbackType)
{
	if (*savedCallback)
	{
		//IwTrace(ODK, ("Unregistering Callback"));

		s3eEdkCallbacksUnRegister(
				S3E_EXT_ODK_HASH,
				S3E_ODK_CALLBACKS_MAX,
				callbackType,
				*savedCallback,
				NULL);
		*savedCallback = NULL;
	}

	*savedCallback = callback;

	//IwTrace(ODK, ("Registering Callback"));

	s3eEdkCallbacksRegister(
			S3E_EXT_ODK_HASH,
			S3E_ODK_CALLBACKS_MAX,
			callbackType,
			*savedCallback,
			NULL,
			S3E_FALSE);
}

void CallbacksSetDeveloperId::RegisterCallbacks(s3eCallback onSuccess, s3eCallback onFailure)
{
	RegisterCallback(onSuccess, &m_onSuccess, S3E_ODK_CALLBACKS_SET_DEVELOPER_ID_ON_SUCCESS);
	RegisterCallback(onFailure, &m_onFailure, S3E_ODK_CALLBACKS_SET_DEVELOPER_ID_ON_FAILURE);
}

void CallbacksSetDeveloperId::OnSuccess()
{
	s3eSetDeveloperIdSuccessEvent event;

	m_dataSetDeveloperIdSuccessEvent = event; //don't send a temp pointer
	s3eEdkCallbacksEnqueue(S3E_EXT_ODK_HASH, S3E_ODK_CALLBACKS_SET_DEVELOPER_ID_ON_SUCCESS, &m_dataSetDeveloperIdSuccessEvent);
}

void CallbacksSetDeveloperId::OnFailure(int errorCode, const std::string& errorMessage)
{
	char buffer[256];
	sprintf(buffer, "OnFailure errorCode=%d errorMessage=%s", errorCode, errorMessage.c_str());
	IwTrace(ODK, (buffer));

	s3eSetDeveloperIdFailureEvent event;
	event.m_errorCode = errorCode;
	sprintf(event.m_errorMessage, "%s", errorMessage.c_str());

	m_dataSetDeveloperIdFailureEvent = event; //don't send a temp pointer
	s3eEdkCallbacksEnqueue(S3E_EXT_ODK_HASH, S3E_ODK_CALLBACKS_SET_DEVELOPER_ID_ON_FAILURE, &m_dataSetDeveloperIdFailureEvent);
}

File: Marmalade/MarmaladeODK/source/android/CallbacksSetDeveloperId.java
/*
 * Copyright (C) 2012, 2013 OUYA, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package tv.ouya.sdk.marmalade;


import android.util.Log;


public class CallbacksSetDeveloperId {

	private final String LOG_TAG  = CallbacksSetDeveloperId.class.getSimpleName();

	public native void CallbacksSetDeveloperIdOnSuccess();
	public native void CallbacksSetDeveloperIdOnFailure(int errorCode, String errorMessage);

	public void onSuccess() {
		Log.i(LOG_TAG, "onSuccess");
		CallbacksSetDeveloperIdOnSuccess();
	}

	public void onFailure(final int errorCode, final String errorMessage) {
		Log.i(LOG_TAG, "onFailure: errorCode=" + errorCode + " errorMessagee=" + errorMessage);
		CallbacksSetDeveloperIdOnFailure(errorCode, errorMessage);
	}
}